tests/suites/test_suite_psa_crypto_se_driver_hal_mocks.function

```diff
@@ -36,6 +44,26 @@ static void mock_teardown( void )
     memset( &mock_export_data, 0, sizeof( mock_export_data ) );
     memset( &mock_allocate_data, 0, sizeof( mock_allocate_data ) );
     memset( &mock_destroy_data, 0, sizeof( mock_destroy_data ) );
+    memset( &mock_generate_data, 0, sizeof( mock_generate_data ) );
+}
+
+static psa_status_t mock_generate( psa_drv_se_context_t *drv_context,
+                                   psa_key_slot_number_t key_slot,
+                                   const psa_key_attributes_t *attributes,
+                                   uint8_t *pubkey,
+                                   size_t pubkey_size,
+                                   size_t *pubkey_length )
+{
+    (void) drv_context;
+    (void) pubkey;
+    (void) pubkey_length;
+
+    mock_generate_data.called++;
+    mock_generate_data.key_slot = key_slot;
+    mock_generate_data.attributes.core.type = attributes->core.type;
+    mock_generate_data.pubkey_size = pubkey_size;
```

Comment:
Just confirming, it's not required as part of the driver contract to set pubkey or pubkey_length. Correct? Nothing in core uses those?

If nothing in core uses those, what's the point of having them other than we expect we might want to use them in the future and want API consistency between this and the next driver model version?

Reply:
It's a bit YAGNI as per our current implementation. However, we feel we need a stable PSA Crypto API and must allow for this capability in the API even if not implemented yet.